File: backend/src/entities/AdminSetting.ts
import { Entity, PrimaryColumn, Column } from "typeorm";

@Entity("AdminSettings")
export class AdminSetting {
  @PrimaryColumn({ type: "nvarchar", length: 50, name: "key" })
  key!: string;

  /** NVARCHAR(MAX) — most settings are short (a URL, a flag, an email), but a
   * few (e.g. FABRIX_OPENAPI_TOKEN's encrypted value) are large signed tokens
   * that would silently overflow a fixed-length column. See migration
   * 1960000000000-WidenAdminSettingValue for the column widen. */
  @Column({ type: "nvarchar", length: "MAX" })
  value!: string;
}

File: backend-py/app/models/admin_setting.py
"""Port of `backend/src/entities/AdminSetting.ts`.

Plain key/value settings table — `value` is `NVARCHAR(MAX)` (widened by
migration `1960000000000-WidenAdminSettingValue`) since a few settings (e.g.
`FABRIX_OPENAPI_TOKEN`'s encrypted value) are large signed tokens.

TODO(phase N): a typed settings-service layer (SMTP/FabriX/Claude/SFTP) reads
and writes rows here via `app.security.secret_cipher` for anything sensitive —
not built in this phase, just the table itself.
"""

from __future__ import annotations

from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, nvarchar_max
from sqlalchemy.dialects import mssql


class AdminSetting(Base):
    __tablename__ = "AdminSettings"

    key: Mapped[str] = mapped_column(mssql.NVARCHAR(50), primary_key=True)
    value: Mapped[str] = nvarchar_max()
